packages/cap-config-writer/src/cap-config/package-json.ts

```diff
@@ -0,0 +1,82 @@
+import { coerce, gte } from 'semver';
+import { getCapCustomPaths } from '@sap-ux/project-access';
+import type { Package } from '@sap-ux/project-access';
+
+const minCdsVersion = '6.8.2';
+
+/**
+ * Ensure a minimum version of @sap/cds in dependencies.
+ *
+ * @param packageJson - the parsed package.json
+ */
+export function ensureMinCdsVersion(packageJson: Package): void {
+    if (!hasMinCdsVersion(packageJson)) {
+        packageJson.dependencies ??= {};
+        packageJson.dependencies['@sap/cds'] = `^${minCdsVersion}`;
+    }
+}
+
+/**
+ * Enable workspaces for app/* folders.
+ *
+ * @param basePath - root path of the CAP project, where package.json is located
+ * @param packageJson - the parsed package.json
+ */
+export async function enableWorkspaces(basePath: string, packageJson: Package): Promise<void> {
+    const { appWorkspace, workspaceEnabled } = await getWorkspaceInfo(basePath, packageJson);
+    if (!workspaceEnabled) {
+        packageJson.workspaces ??= [];
+        if (Array.isArray(packageJson.workspaces)) {
```

Comment:
That if is redundant, you are checking it already in `getWorkspaceInfo`

Reply:
Changed this implementation, also consider now workspace config done like
```
workspaces: {
  packages: []
}
```

This is used e.g. for yarn workspaces.